src/test/codegen/issue-83623-SIMD-PartialEq.rs

```diff
@@ -0,0 +1,45 @@
+// This test checks that jumps generated by logical operators can be optimized away
+
+// compile-flags: -O3
+// only-64bit
```

Comment:
This test _might_ fail for some of the 64-bit targets, and _may_ need to be limited to just `x86_64`, but lets see if this works out, first.